metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py

```diff
@@ -45,3 +52,48 @@ def get_tables(native_query: str) -> List[str]:
         from_index = from_index + 1
 
     return tables
+
+
+def parse_custom_sql(
+    ctx: PipelineContext,
+    query: str,
+    schema: Optional[str],
+    database: Optional[str],
+    platform: str,
+    env: str,
+    platform_instance: Optional[str],
+) -> Optional["SqlParsingResult"]:
+
+    logger.debug("Using sqlglot_lineage to parse custom sql")
+
+    sql_query = remove_special_characters(query)
+    logger.debug(f"Parsing sql={sql_query}")
+
+    parsed_result: Optional["SqlParsingResult"] = None
+    try:
+        schema_resolver = (
```

Comment:
some of this logic looks like it was copy-pasted from elsewhere

can we instead extract it into a common helper method?

Reply:
I added it in sqlglot_lineage as one additional method